server/src/com/cloud/api/ApiResponseGsonHelper.java

```diff
@@ -71,4 +83,19 @@ public boolean shouldSkipField(FieldAttributes f) {
             return false;
         }
     }
+
+    private static class LogExclusionStrategy extends ApiResponseExclusionStrategy implements ExclusionStrategy {
```

Comment:
The existing LoggingExlusionStrategy is for agent commands/workjob etc. But it won't work for API responses. If you look at the code, exclusion strategy for API response log depends on ApiResponseExclusionStrategy as well for exclusion based on roles - user/admin.